03_Modules/Certificates/src/module/api.ts

```diff
@@ -436,6 +526,279 @@ export class CertificatesModuleApi
+            countryName,
+            validBefore,
+            signatureAlgorithm,
+          );
+        }
+        existingInstalledCertificate.certificateId = existingCertificate.id;
+        existingInstalledCertificate =
+          await existingInstalledCertificate.save();
+      }
+    } else {
+      // create new certificate record
+      const newCertificate: Certificate = await this.createNewCertificate(
+        certificate,
+        serialNumber,
+        issuerName,
+        organizationName,
+        commonName,
+        countryName,
+        validBefore,
+        signatureAlgorithm,
+      );
+      existingInstalledCertificate = new InstalledCertificate();
+      existingInstalledCertificate.stationId = messageQuerystring.identifier;
+      existingInstalledCertificate.certificateId = newCertificate.id;
+      existingInstalledCertificate.certificateType =
+        uploadExistingCertificate.certificateType;
+      existingInstalledCertificate = await existingInstalledCertificate.save();
+    }
+    return existingInstalledCertificate;
+  }
+
+  @AsDataEndpoint(
+    Namespace.RegenerateExistingCertificate,
+    HttpMethod.Post,
+    IMessageQuerystringSchema,
+    RegenerateInstalledCertificateSchema,
+  )
+  async regenerateExistingCertificate(
+    request: FastifyRequest<{
+      Body: RegenerateExistingCertificate;
+      Querystring: IMessageQuerystring;
+    }>,
+  ): Promise<InstalledCertificate> {
+    const installedCertificateId = request.body.installedCertificateId;
+    const stationId = request.query.identifier;
+    this._logger.info(
+      `Regenerating existing certificate ${installedCertificateId} for charger ${stationId}`,
+    );
+    const existingInstalledCertificate =
+      await this._module.installedCertificateRepository.readOnlyOneByQuery({
+        where: {
+          id: installedCertificateId,
+          stationId: stationId,
+        },
+      });
+    if (!existingInstalledCertificate) {
+      throw new Error('Installed certificate not found');
+    }
+    const existingCertificateRecord =
+      await existingInstalledCertificate.$get('certificate');
+    if (!existingCertificateRecord) {
+      throw new Error('Certificate not found');
+    }
+    const fileId = existingCertificateRecord.certificateFileId;
+    if (!fileId) {
+      throw new Error('Certificate file not found');
+    }
+    const existingCertificateBuffer = await this._fileAccess.getFile(fileId);
+    const existingCertificate = existingCertificateBuffer.toString();
+    const [newCertificatePem, _newPrivateKeyPem] = generateCertificate(
+      existingCertificateRecord,
+      this._logger,
+      undefined,
+      existingCertificate,
+    );
+    const newCertificateHash = this.getCertificateHash(newCertificatePem);
+    let newCertificateRecord = new Certificate();
+    newCertificateRecord.serialNumber = moment().valueOf();
+    newCertificateRecord.issuerName = existingCertificateRecord.issuerName;
```

Comment:
The issuerName of the new cert should be the subjectName of the existing certificate.